### ISSUE TITLE
Sklearn pipeline not working with mitigation

### ISSUE THREAD

**Issue by smrnvdhy** [fairlearn/fairlearn#1041]
#### Describe the bug
sklearn pipeline not working with mitigation technique.

#### Steps/Code to Reproduce
I'm following the example given in [Passing pipelines to mitigation techniques](https://fairlearn.org/main/auto_examples/plot_mitigation_pipeline.html)

#### Expected Results
No error (works fine when pipeline is broken down. suspecting issue is with preprocessing)
![image](https://user-images.githubusercontent.com/61920451/157914832-7ec1dad9-d66f-447f-8c7c-af882742b2c4.png)

#### Actual Results

![image](https://user-images.githubusercontent.com/61920451/157914968-1205fee5-18ef-4e2c-8a3e-a1995b4a3e88.png)

#### Versions

![image](https://user-images.githubusercontent.com/61920451/157915354-9ade516f-9fe8-4f1a-9c85-52607aac0114.png)

**Comment by romanlutz** [comment]
Thanks for reporting this issue!

@smrnvdhy could you provide information on your `X_train_processed`? It looks like one of the features, `Self-emp-not-inc`, is a string rather than a numeric value. That may be the reason this isn't working, but it hard to say without knowing more.

**Comment by smrnvdhy** [comment]
@romanlutz Oh, I'm sorry I was unclear. The error is thrown when the code example in step-to-reproduce section is followed verbatim: `X_train_processed` is not involved in this error.

I included the object `X_train_processed`, which comes from applying preprocessing to `X_train`, to show that the mitigation algorithm works fine when preprocessing and model fittings are done separately (not in pipeline).

**Comment by romanlutz** [comment]
Oh sorry, I missed the link. This is from our website and therefore actually executed to generate that page. It definitely works, otherwise you'd see an error on the webpage. That said, it may still fail locally for you, of course, because your environment is not identical.

What's your Fairlearn version? You might notice that this page was added only on "main", i.e., it's not present in all released versions including the latest (0.7.0). I am not 100% certain, but I vaguely recall there being an issue like this before. It might be #859 or #309 (although the latter was a long time ago and is definitely included in the latest release...).

We are planning to do a release soon (it was targeted for end of March, but it looks like it will happen in April instead), so you can upgrade at that point. In the meanwhile you can try doing `pip install` from a `git clone`d version of Fairlearn. Let us know if that works for you!

**Comment by smrnvdhy** [comment]
I completely missed main vs latest version. My version is indeed 0.7.0. I'll wait for the release!

On a side note, do you have any recommendation for multiclass/multilabel classification? I understand fairness criteria isn't as well defined as in binary classification. I like the idea presented in [adversarial debiasing (#785)](https://github.com/fairlearn/fairlearn/issues/785) which seems to generalize for any task.

**Comment by romanlutz** [comment]
Sounds good. 

Search for "multiclass" on this page to find the FAQ section for it: https://fairlearn.org/main/faq.html

There's an open issue for the paper you referenced, and I believe someone is actively working on it. Just search the paper title in our issues and you should find it. In the interest of keeping issues clear I suggest you open an new one if you have a new feature request on that topic. Thanks!

**Comment by adrinjalali** [comment]
I think this seems to be solved already, so closing the issue.

**Comment by andreaskopp** [comment]
Is it solved? I have exactly the same issue trying the with Fairlearn version 0.7.0 using the demo notebook https://fairlearn.org/main/auto_examples/plot_mitigation_pipeline.html

Can you share the solution, please?

**Comment by adrinjalali** [comment]
Could you please share a minimal reproducing example (something we can copy/paste to reproduce your error) here, and test it against the latest upstream? You can install the latest upstream using

```
pip install git+https://github.com/fairlearn/fairlearn.gi
```

**Comment by riedgar-ms** [comment]
I can repro in an AzureML notebook. Must be something in the environment as compared to the build.

**Comment by adrinjalali** [comment]
sklearn version?

**Comment by adrinjalali** [comment]
Should it not be? 

If you could post an environment file/requirement file + the code reproducing the error it'd be great.

**Comment by andreaskopp** [comment]
Yes, categorical should be correct for 'workclass'

Regarding, repro code. It is the original demo notebook: https://fairlearn.org/main/auto_examples/plot_mitigation_pipeline.html

**Comment by riedgar-ms** [comment]
The full output of `pip freeze`:

[aml-vm.txt](https://github.com/fairlearn/fairlearn/files/8870668/aml-vm.txt)

**Comment by riedgar-ms** [comment]
And `conda list --explicit`:

[conda-explicit.txt](https://github.com/fairlearn/fairlearn/files/8870730/conda-explicit.txt)

**Comment by riedgar-ms** [comment]
I can also repro locally with that freeze output. This time `sklearn.show_versions()` is:
```
System:
    python: 3.8.13 (default, Mar 28 2022, 06:59:08) [MSC v.1916 64 bit (AMD64)]
executable: C:\Users\riedgar\AppData\Local\Continuum\Miniconda3\envs\fairlearn-nb\python.exe
   machine: Windows-10-10.0.19044-SP0

Python dependencies:
      sklearn: 1.1.1
          pip: 21.2.2
   setuptools: 61.2.0
        numpy: 1.22.4
        scipy: 1.8.1
       Cython: None
       pandas: 1.4.2
   matplotlib: 3.5.2
       joblib: 1.1.0
threadpoolctl: 3.1.0

Built with OpenMP: True
```

**Comment by riedgar-ms** [comment]
Simpler way to repro the 'NaN' error:
[simpler-nan-repro.txt](https://github.com/fairlearn/fairlearn/files/8871596/simpler-nan-repro.txt)

I'm wondering if `check_X_y()` has changed behaviour of late. And whether Fairlearn is being overly eager in its validations.

But also: why haven't our tests caught this?

**Comment by riedgar-ms** [comment]
I have Good News, and I have Bad News:

- This was [fixed almost a year ago](https://github.com/fairlearn/fairlearn/commit/a68e74ec90ce99f00446b0b8fe53203f61f640b3)
- We still haven't release v0.8.0

**Comment by riedgar-ms** [comment]
@andreaskopp please can you try installing Fairlearn from GitHub:

```bash
pip install git+https://github.com/fairlearn/fairlearn.git
```

That should fix the issue.

**Comment by adrinjalali** [comment]
That's why I said to try the latest lol

So the issue should stay closed as fixed, there's no action to be done for this issue.

**Comment by riedgar-ms** [comment]
Agreed.

**Comment by andreaskopp** [comment]
> @andreaskopp please can you try installing Fairlearn from GitHub (as @adrinjalali suggested):
> 
> ```shell
> pip install git+https://github.com/fairlearn/fairlearn.git
> ```
> 
> That should fix the issue.

Yes, done and it worked! Thanks for your quick support! Would be important to have this as an official release soon.
